client/src/main/java/org/apache/rocketmq/client/impl/factory/MQClientInstance.java

```diff
@@ -1044,9 +1044,14 @@ public FindBrokerResult findBrokerAddressInSubscribe(
             found = brokerAddr != null;
 
             if (!found && !onlyThisBroker) {
-                Entry<Long, String> entry = map.entrySet().iterator().next();
-                brokerAddr = entry.getValue();
-                slave = entry.getKey() != MixAll.MASTER_ID;
+                if (brokerId == 1 && map.size() > 2) {
+                    brokerAddr = map.get(2L);
```

Comment:
why just choose the concrete index 2 while not 3, 4 or other if size > 2? Is there any optimized space?

Reply:
I agree with u, I'will try to find a more elegant way.